server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1554,7 +1584,20 @@ public void onFailure(Exception e) {
                     shardGenerations
                 )
             ));
-        } catch (Exception e) {
+        }
+
+        @Override
+        public void onRejection(Exception e) {
+            if (e instanceof EsRejectedExecutionException esre && esre.isExecutorShutdown()) {
+                logger.debug("failing finalization of {} due to shutdown", snapshot);
```

Comment:
Nice extra handling.

To verify my understanding, and make a note of the behavior change: previously we'd log a warning about failing due to shutdown; now we're making that quiet under debug. That seems reasonable.

Reply:
Previously we didn't fork here, we just ran all this on the calling thread, so rejection was not a thing that could happen.